make/common/MakeBase.gmk

```diff
@@ -140,6 +140,17 @@ FindAllReposRel = \
 
 ################################################################################
 
+# If reproducible builds are enabled then set SOURCE_DATE_ISO_8601 string variable
+ifeq ($(ENABLE_REPRODUCIBLE_BUILD), true)
+  SOURCE_DATE_ISO_8601 := $(shell $(DATE) --utc --date="@$(SOURCE_DATE_EPOCH)" +"%Y-%m-%dT%H:%M:%SZ" 2> /dev/null)
```

Comment:
This should be set up by the configure script. Look for where we are currently doing the SOURCE_DATE_EPOCH stuff.

Also consider the possibility to *just* use this variable, and pass it on to jar if it is present, and not if it's missing. That way you can get rid of the redundant boolean variable.

Reply:
@magicus I looked at that, but the problem is SOURCE_DATE_EPOCH is set in InitSupport.gmk depending on whether SOURCE_DATE=="updated" : https://github.com/openjdk/jdk/blob/739769c8fc4b496f08a92225a12d07414537b6c0/make/InitSupport.gmk#L315
I also couldn't add it in InitSupport.gmk because that marco is not included from every place SetupJarArchive is resolved.
Thoughts?

Reply:
Oh, that's ... interesting. (I'm pretty sure I wrote that code myself :))

I still think it would be good to keep the new code close to the old. If we set SOURCE_DATE to "updated", I think that should reflect in SOURCE_DATE_ISO_8601 as well. Maybe it does by the current design, but if it does, it could be more obvious.

I'm sorry I don't have any ready-made suggestion. :( I'm really on vacation now and can't really dive into this, so if you can't find any better solution, then this'll have to do.

Reply:
But I think the code in InitSupport will be executed always; Init.gmk is our "bootstrapper" / "trampoline" which wraps all calls to make (and InitSupport.gmk contains gory implementation details of Init.gmk).

Reply:
SOURCE_DATE_EPOCH is initialized and exported in InitSupport.gmk so it's always available in the environment. We did this because we want various tools to pick this variable up from the environment, as this is a commonly expected variable name for doing so. The new variable SOURCE_DATE_ISO_8601 is a variant with a different format and there is no standard for reading this from the environment, so it's not as obvious that we should just export it the same way. 

On the other hand, we do not want to execute a shell expression every time we import MakeBase.gmk, so I would still vote for doing this in InitSupport.gmk and export SOURCE_DATE_ISO_8601.

Reply:
@erikj79 Interesting, I tried adding it to InitSupport.gmk, but it wasn't always set... odd, I will revisit, I probably did something wrong...! will give that another try, thanks

Reply:
@magicus don't worry, enjoy your holiday, I think @erik has pointed out it should work, so I probably made a mistake when I tried it...

Reply:
Moved export SOURCE_DATE_ISO_8601 to InitSupport as suggested, found the problem as to why it didn't work for me, I was missing a "$" !